File: pkg/resources/audit-log.go
package resources

import (
	"github.com/apibrew/apibrew/pkg/model"
	"github.com/apibrew/apibrew/pkg/resources/special"
	"github.com/apibrew/apibrew/pkg/service/annotations"
)

var AuditLogResource = &model.Resource{
	Name:      "AuditLog",
	Namespace: "system",
	SourceConfig: &model.ResourceSourceConfig{
		DataSource: "system",
		Entity:     "audit_log",
	},
	Immutable: true,
	Properties: map[string]*model.ResourceProperty{
		"id":      special.IdProperty,
		"version": special.VersionProperty,
		"namespace": {
			Length:   256,
			Type:     model.ResourceProperty_STRING,
			Required: true,
		},
		"resource": {
			Length:   256,
			Type:     model.ResourceProperty_STRING,
			Required: true,
		},
		"recordId": {
			Length:   256,
			Type:     model.ResourceProperty_STRING,
			Required: true,
			Annotations: map[string]string{
				annotations.SourceDef: "record_id",
			},
		},
		"time": {
			Type:     model.ResourceProperty_TIMESTAMP,
			Required: true,
		},
		"username": {
			Type:     model.ResourceProperty_STRING,
			Required: true,
		},
		"operation": {
			Type:     model.ResourceProperty_ENUM,
			Required: true,
			EnumValues: []string{
				"CREATE",
				"UPDATE",
				"DELETE",
			},
		},
		"properties": {
			Type:     model.ResourceProperty_OBJECT,
			Required: false,
		},
		"annotations": special.AnnotationsProperty,
	},
	Annotations: map[string]string{
		annotations.OpenApiGroup:     OpenApiInternal,
		annotations.BypassExtensions: annotations.Enabled,
	},
}

File: sdk/nodejs/src/model/audit-log.ts

export interface AuditLog {
    id: string
    version: number
    namespace: string
    resource: string
    recordId: string
    time: string | Date
    username: string
    operation: Operation
    properties?: object
    annotations?: { [key: string]: string }
}

export const AuditLogEntityInfo = {
    namespace: "system",
    resource: "AuditLog",
    restPath: "system-auditlog",
}

export enum Operation {
    CREATE = "CREATE",
    UPDATE = "UPDATE",
    DELETE = "DELETE",
}

export const AuditLogResource = {
  "name": "AuditLog",
  "namespace": {
    "name": "system"
  },
  "properties": {
    "annotations": {
      "type": "MAP",
      "item": {
        "type": "STRING"
      },
      "exampleValue": {
        "CheckVersion": "true",
        "CommonType": "testType",
        "IgnoreIfExists": "true"
      },
      "title": "Annotations",
      "description": "The annotations of the resource/record. It contains information about the resource/record. For example, it can contain information about the UI representation of the resource/record.",
      "annotations": {
        "SpecialProperty": "true"
      }
    },
    "id": {
      "type": "UUID",
      "required": true,
      "immutable": true,
      "exampleValue": "a39621a4-6d48-11ee-b962-0242ac120002",
      "description": "The unique identifier of the resource. It is randomly generated and immutable.",
      "annotations": {
        "PrimaryProperty": "true",
        "SpecialProperty": "true"
      }
    },
    "namespace": {
      "type": "STRING",
      "required": true,
      "length": 256
    },
    "operation": {
      "type": "ENUM",
      "required": true,
      "enumValues": [
        "CREATE",
        "UPDATE",
        "DELETE"
      ]
    },
    "properties": {
      "type": "OBJECT"
    },
    "recordId": {
      "type": "STRING",
      "required": true,
      "length": 256,
      "annotations": {
        "SourceDef": "record_id"
      }
    },
    "resource": {
      "type": "STRING",
      "required": true,
      "length": 256
    },
    "time": {
      "type": "TIMESTAMP",
      "required": true
    },
    "username": {
      "type": "STRING",
      "required": true
    },
    "version": {
      "type": "INT32",
      "required": true,
      "defaultValue": 1,
      "exampleValue": 1,
      "title": "Version",
      "description": "The version of the resource/record. It is incremented on every update.",
      "annotations": {
        "AllowEmptyPrimitive": "true",
        "SpecialProperty": "true"
      }
    }
  },
  "immutable": true,
  "annotations": {
    "BypassExtensions": "true",
    "OpenApiGroup": "internal"
  }
} as unknown

